data/changelog/posts/ocaml-lsp/2025-08-20-ocaml-eglot-brings-lsp-support-to-emacs.md

```diff
@@ -0,0 +1,94 @@
+
+**Quick start**: [Install `ocaml-lsp-server`, add `ocaml-eglot` to your Emacs config](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#installation), and get the same OCaml development experience with less configuration.
+
+## What is ocaml-eglot?
+
+`ocaml-eglot` connects Emacs to `ocaml-lsp-server` using the Language Server Protocol, providing a standardized way to get OCaml language support.
+
+Since the recent versions of Emacs (29), `eglot`, an LSP client, has been shipped with Emacs. However, `merlin.el` provides more features than LSP (which is designed to be generic), so relying solely on the features of LSP and `eglot` would limit functionality. Thus, we extended the LSP server to support more features, and `ocaml-eglot` allows you to benefit from these features in Emacs.
+
+`ocaml-eglot` is a minor mode. It therefore works in conjunction with a major mode to edit Caml code. Examples of major modes include `tuareg`, `caml-mode` and the recent `neocaml`.
+
+## Who Should Use ocaml-eglot?
+
+**Use ocaml-eglot if you:**
+- Are starting fresh with OCaml in Emacs
+- Want simplified configuration with automatic setup
+- Use multiple editors and want consistent OCaml support
+- Want access to project-wide search and rename features
+- Want to rely on an actively maintained project that evolves over time
+
+**Stick with `merlin.el` if:**
+- Your current setup is working perfectly and heavily customized
+- You prefer direct Merlin communication
+
+For the moment, we don't plan to provide any special support for `merlin.el` -- unless we receive a lot of requests.
+
+## Key Benefits
+
+- **Simplified setup**: Install package, add config, start coding
+- **Same features**: All `merlin.el` functionality with identical keybindings
+- **Enhanced capabilities**: Project-wide rename, type-based search, automatic formatting
+- **Platform integration**: Works seamlessly with opam, dune, ocamlformat, and other OCaml tools
+
+## Getting Started
+
+Follow the [installation instructions in the ocaml-eglot README](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#installation).
+
+For a detailed list of features, see [the ocaml-eglot README](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#features).
+
+## Migration from merlin.el
+
+1. Install: `opam install ocaml-lsp-server ocamlformat`
+2. Replace your Merlin configuration with the `ocaml-eglot` setup above
+3. Restart Emacs
+
+Your existing keybindings work immediately.
+
+## Enhanced Features Examples
+
+**Type-based search**: Find functions by signature
+```
+Search for: string -> int option
+Result: String.to_int, Int.of_string, etc.
+```
+
+**Project-wide rename** (OCaml 5.2+):
+```bash
+# Build index first
+dune build @ocaml-index
+# Then use ocaml-eglot-rename in Emacs
+```
+
+**Type search**: Find functions by input/output types
```

Comment:
It is called `search by polarity` and I was suggesting to remove the spotlight on that feature (since `search by type` is better and search by polarity does not support parametrized types properly).

Reply:
But I still thinking that we can have an unformal enumeration (without details) of features from `ocaml-eglot`, a link to the README instead of the cherry picking of two features. WDYT?